akka-docs/src/test/java/jdocs/tutorial_5/DeviceGroup.java

```diff
@@ -139,8 +139,14 @@ private void onTerminated(Terminated t) {
   //#query-added
 
   private void onAllTemperatures(RequestAllTemperatures r) {
+    // since Java collections are mutable, we want to avoid sharing them between actors (since multiple Actors (threads)
+    // modifying the same mutable data-structure is not safe), and perform a defensive copy of the mutable map:
+    //
+    // Feel free to use your favourite immutable data-structures library with Akka in Java applications!
+    HashMap<ActorRef, String> actorToDeviceIdCopy = new HashMap<>(this.actorToDeviceId);
```

Comment:
If this is in the docs i'd type this to Map

Reply:
java makes me sad

Reply:
![](https://media3.giphy.com/media/XCB6lEBSSw8oM/giphy.gif)